test/suite/test_checkpoint_snapshot02.py

```diff
@@ -264,7 +302,13 @@ def test_checkpoint_snapshot_with_txnid_and_timestamp(self):
         self.assertGreater(inconsistent_ckpt, 0)
         self.assertGreaterEqual(keys_removed, 0)
 
-        simulate_crash_restart(self, "RESTART", "RESTART2")
+        if self.restart == True:
+            simulate_crash_restart(self, "RESTART", "RESTART2")
+        else:
+            #Take a backup and restore it.
+            self.take_full_backup(self.backup_dir, self.backup_dir2)
+            self.reopen_conn(self.backup_dir2)
```

Comment:
Can we extract the if/else block above into a separate method to avoid duplication?

Reply:
Added a new function to remove duplicate code.